magento/magento2
# Upgraded phpstan dependency to 1.1

### Description (*)
PHPstan version 1.0 got released earlier this week: https://phpstan.org/blog/phpstan-1-0-released
This PR changes the dependency constraints for phpstan.
This also changes the deprecated `excludes_analyse` config to `excludePaths`.

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/650


### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/34604

### Manual testing scenarios (*)
1. Run phpstan on various php files to see if it still works without issues, for example:
```
./vendor/bin/phpstan analyse --level 1 --no-progress --memory-limit=4G --configuration ./dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/phpstan.neon setup/src/Magento/Setup/Test/Unit/Console/Style/MagentoStyleTest.php
```
(this command comes from https://github.com/magento/magento2/pull/30581#issuecomment-780108973, and also see: https://github.com/magento/magento2/blob/2.4-develop/dev/tests/static/framework/Magento/TestFramework/CodingStandard/Tool/PhpStan.php#L81-L92)

### Questions or comments
Should we still allow version 0.12.x of phpstan to get installed as well, or is that not necessary?

### Contribution checklist (*)
